### DISCUSSION TITLE
Table Block Showing Rows which are deleted by the databse.

### DISCUSSION THREAD

**Discussion by thering1975TG** [Budibase/budibase#13267] [Help] [answered]
Hi

Choose to put this here rather than a bug as it may be something i am doing wrong

So i have a form which is completed by users, this creates a new row in the internal CouchDB database.

I was doing some tests due to another issue i faced and solved. These rows were created in the Database fine. So as these were test entries i selected the 2 rows in the backend Database screen and deleted. No issues there the rows were deleted.

I was checking something in the front end just now and noticed that the web page with a  TableBlock to display the data and side panel of this form was still showing these 2 deleted rows which do not exist any more.

I cleared cache and checked in a new browser and those rows were still being displayed in the front end view.

I checked in the design tab in case i missed a refresh setting or something and in the design view the table block section was correct i.e those 2 rows that were deleted were not shown in the design page, if i previewed it the rows were also not displayed, so this is correct

So i decided to do a last test and unpublish and republish the app, as soon as i did this the 2 rows were now NOT shown in the frontend webui as it should be.

So why is the table block updating correctly when adding a new row from the front end, but if i delete a row in the backend it does not refresh and these rows that do not exist are still showing?

**Comment by ConorWebb96** [comment, marked as answer]
I can see you have unpublished and re-published and this seemed to fix your issue. An explanation on why this happened is below.

This sounds like you deleted them whilst in the preview screen. This likely means your published database and dev database are out of sync. Best to always delete rows from production as this keeps everything in-sync.

I hope this answers your query.
